### PR TITLE
Improve Integration Test Stability

### PR DESCRIPTION
**Pull request by jhbertra**
- [x] Add a new option to `marlowe-tx` for waiting a number of blocks before confirming a transaction
- [x] Set the option to `2` for the integration tests.

### CODE REVIEW

File: marlowe-integration/src/Test/Integration/Marlowe/Local.hs [hunk 1]
```diff
@@ -285,6 +285,7 @@ withLocalMarloweRuntime' MarloweRuntimeOptions{..} test = withRunInIO \runInIO -
     webPort <- liftIO $ randomRIO (4000, 5000)
     manager <- liftIO $ newManager defaultManagerSettings
 
+    let submitConfirmationBlocks = 2
```
Comment (bwbush) [inline marlowe-integration/src/Test/Integration/Marlowe/Local.hs:288]: If we still have troubles with the integration tests on the fast network, we can increase this further.

File: marlowe-runtime/marlowe-tx/Main.hs [hunk 2]
```diff
@@ -244,6 +247,14 @@ getOptions = execParser $ info (helper <*> parser) infoMod
       , help "The logging configuration JSON file."
       ]
 
+    submitConfirmationBlocksParser = option (BlockNo <$> auto) $ mconcat
+      [ long "submit-confirmation-blocks"
```
Comment (bwbush) [inline marlowe-runtime/marlowe-tx/Main.hs:251]: Nice!

File: marlowe-integration/src/Test/Integration/Marlowe/Local.hs (hunk 1 above)
Comment (jhbertra) [reply marlowe-integration/src/Test/Integration/Marlowe/Local.hs:288]: Yeah, this is our knob. This is an "optimistic" initial value

File: marlowe-runtime/marlowe-tx/Main.hs [hunk 3]
```diff
@@ -244,6 +247,14 @@ getOptions = execParser $ info (helper <*> parser) infoMod
...
+      , value 0
```
Comment (bwbush) [inline marlowe-runtime/marlowe-tx/Main.hs:252]: Yes, the default should be zero because some clients will want to tune this themselves.

Comment (jhbertra) [reply marlowe-runtime/marlowe-tx/Main.hs:252]: It also preserves the current behaviour by default

**Comment by jhbertra** [comment]
Reopen to try and create the jobset on Hydra

File: marlowe-runtime/tx/Language/Marlowe/Runtime/Transaction/Submit.hs [hunk 4]
```diff
@@ -61,18 +63,38 @@ doSubmit SubmitJobDependencies{..} tellStatus era tx= do
...
-          , recvMsgRollForward = \_ point _ -> case point of
+          , recvMsgRollBackward = \_ _ -> pure clientIdleAwaitConfirmation
+          , recvMsgRollForward = \_ point tip -> case (point, tip) of
+              (At block, At tipBlock) -> pure $ clientIdleAwaitMaturity block tipBlock
+              _ -> error "chainseekd rolled forward to genesis"
+          , recvMsgWait = threadDelay 100_000 $> SendMsgPoll clientNextAwaitConfirmation
+          }
+
+        clientIdleAwaitMaturity block tipBlock
+          | blockNo tipBlock - blockNo block >= submitConfirmationBlocks = SendMsgDone block
```
Comment (bwbush) [inline marlowe-runtime/tx/Language/Marlowe/Runtime/Transaction/Submit.hs:86]: Should `WaitForTx` allow the global default confirmation to be overridden by clients?

Comment (jhbertra) [reply marlowe-runtime/tx/Language/Marlowe/Runtime/Transaction/Submit.hs:86]: Perhaps, we could add this as an option to the `Submit` command

Comment (bwbush) [reply marlowe-runtime/tx/Language/Marlowe/Runtime/Transaction/Submit.hs]: Let's hold off, since we might deprecate `Submit`.

**Review by bwbush — APPROVED** [review]

**Pull request merged**